engine-standalone-storage/src/sync/types.rs

```diff
@@ -106,7 +114,13 @@ pub enum TransactionKind {
 }
 
 #[derive(BorshDeserialize, BorshSerialize)]
-struct BorshableTransactionMessage<'a> {
+enum BorshableTransactionMessage<'a> {
+    V1(BorshableTransactionMessageV1<'a>),
+    V2(BorshableTransactionMessageV2<'a>),
+}
```

Comment:
FYI, this is not compatible with BorshableTransactionMessageV1. If you try to deserialize a legacy tx it will fail, because it is expecting the enum type on the first byte.

An alternative solution is making a custom implementation of Borsh(De)Serialize.

Reply:
If the plan is to drop support to legacy types, then we can drop the enum completely!

Reply:
The custom deserialization logic is [here](https://github.com/aurora-is-near/aurora-engine/pull/575/files#diff-ad2bf34db0628958ff395ae9b757d564469163a314630a681e8bbb5be2a1b5b7R53). Notice we first try to deserialize the enum, then if it fails we fall back on the V1 serialization. Also note that `BorshableTransactionMessage` is private, so we don't need to worry about others not knowing about this special logic.

I decided to do it this way because I wanted to avoid having to do a DB migration on the engine (and now also refiner), and I did not think it would be very common to look at transaction data, so the slightly inefficient deserialization logic should not be a big performance problem (transaction data is only read from the DB to do debug tracing, which is not a very frequently used endpoint on average).

But let me know if you think it is worth doing the DB migration instead of (or perhaps in addition to) having the custom deserialization logic.

Reply:
I missed that fallback part! It seems ok to me this way.